kinto/core/initialization.py

```diff
@@ -412,10 +412,60 @@ def __call__(self, event):
         return not resource or not self.resources or resource in self.resources
 
 
+class EventResourceIdFilter:
+    def __init__(self, resource_ids, config):
+        self.resource_ids = resource_ids
+
+    def phash(self):
+        return 'for_resource_ids = {}'.format(','.join(self.resource_ids))
+
+    def __call__(self, event):
+        request = event.request
+        uri = event.payload.get('uri')
+        r_name, matchdict = utils.view_lookup(request, uri)
+        """ Special case for Create bucket """
+        # if event.payload.get('action') == 'create' and r_name == 'bucket':
+        # return True
+        for resource_ids in self.resource_ids:
```

Comment:
Maybe you would not need to mock the requests attributes in the other tests that are unrelated to `resource ids` if `utils.view_lookup()` does not get called when `self.resource_ids` is falsy (empty list). You could add `if not self.resource_ids: return` on top and this way this filter is completely bypassed when unused.

(*my 2 cents*)

Reply:
This was helpful, thanks for pointing me in this direction.